### PR TITLE
doc: services: device_mgmt: smp: Add SMP version 2 details

### PR DESCRIPTION
**Pull request by nordicjm**
Adds SMP version 2 error response details to the documentation.

### CODE REVIEW

**Review by de-nordic — DISMISSED** [review]

File: doc/services/device_mgmt/smp_groups/smp_group_0.rst [hunk 1]
```diff
@@ -91,23 +91,44 @@ CBOR data of successful response:
...
+    | "r"              | replying echo string.                                                   |
+    +------------------+-------------------------------------------------------------------------+
+    | "err" -> "group" | :c:enum:`mcumgr_group_t` group of the group-based error code. Only      |
+    |                  | appears if an error is returned when using SMP version 2.               |
+    +------------------+-------------------------------------------------------------------------+
+    | "err" -> "rc"    | contains the index of the group-based error code. Only appears if       |
+    |                  | non-zero (error condition) when using SMP version 2.                    |
+    +------------------+-------------------------------------------------------------------------+
+    | "rc"             | :c:enum:`mcumgr_err_t` only appears if non-zero (error condition) when  |
+    |                  | using SMP version 1.                                                    |
```
Comment (de-nordic) [inline doc/services/device_mgmt/smp_groups/smp_group_0.rst:130]: How are the SMP errors returned with protocol version 2, they have own group now?

Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:130]: Yes, "err" as the key then a map with "group" for group ID and "rc" for error code

Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:130]: I mean the previous protocol error codes, like failed to alloc mem which could also be returned without getting into group specific code.

Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:130]: They will still be returned as version 1 error codes

Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_0.rst:130]: I will clarify the text to show that better

**Review by de-nordic — APPROVED** [review]

**Comment by nordicjm** [comment]
@cfriedt would you be able to merge please?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**